cli/pkg/kctrl/cmd/package/repository/add_or_update.go

```diff
@@ -325,7 +330,9 @@ func (o AddOrUpdateOptions) dryRun() error {
 
 	if tag == "" {
 		packageRepo.Spec.Fetch.ImgpkgBundle.TagSelection = &versions.VersionSelection{
-			Semver: &versions.VersionSelectionSemver{},
+			Semver: &versions.VersionSelectionSemver{
```

Comment:
It seems like code is not ensuring only 1 of the semver or tag in the URL needs to be provided. We can update in the description of the flag which one takes precedence. 

Did you test what happens when both tag and semver are provided. How does Package Repository looks like?